pkg/api/vlabs/validate.go

```diff
@@ -274,14 +277,21 @@ func (a *Properties) Validate() error {
 	return nil
 }
 
-func (a *Properties) ValidateNetworkPolicy() error {
-
+func (a *Properties) validateNetworkPolicy() error {
 	if a.OrchestratorProfile.OrchestratorType != Kubernetes {
 		return nil
 	}
 
 	networkPolicy := a.OrchestratorProfile.KubernetesConfig.NetworkPolicy
```

Comment:
(I think this is a new bit of validation that theoretically should've been added when Calico introduced the KubernetesConfig field.

Basically we need additional type checking of sorts to say `OrchestratorProfile` can only contain `KubernetesConfig` when `OrchestratorProfile.OrchestratorType == Kubernetes`.